core/trino-main/src/test/java/io/trino/operator/aggregation/BenchmarkDecimalAggregation.java

```diff
@@ -108,36 +130,66 @@ public Block benchmarkEvaluateFinal(BenchmarkData data)
         @Param({"10", "1000"})
         private int groupCount = 10;
 
+        @Param({"SHORT", "LONG", "MIXED"})
+        private String decimalSize = "SHORT";
+
+        @Param({"true", "false"})
+        private boolean groupIdsInOrder;
+
         private AggregatorFactory partialAggregatorFactory;
         private AggregatorFactory finalAggregatorFactory;
         private GroupByIdBlock groupIds;
         private Page values;
         private Page intermediateValues;
+        private GroupedAggregator partialAggregator;
 
         @Setup
         public void setup()
         {
             TestingFunctionResolution functionResolution = new TestingFunctionResolution();
-
+            Random random = new Random(343526534);
             switch (type) {
                 case "SHORT": {
                     DecimalType type = createDecimalType(14, 3);
-                    values = createValues(functionResolution, type, type::writeLong);
+                    values = createValues(functionResolution, type, (builder, value) -> {
+                        boolean writeShort = "SHORT".equals(decimalSize) || ("MIXED".equals(decimalSize) && random.nextBoolean());
+                        if (writeShort) {
+                            builder.writeLong(value);
```

Comment:
both cases are long really `writeLong`

Reply:
> yes, but if we use varint those are different cases.

Here `SHORT` vs `LONG` is mixed with Short/Long decimal. This is confusing

Reply:
replaced SHORT, LONG with SMALL, BIG